---
layout: archive
title: Zulip Chat Archive
permalink: /stream/113488-general/topic/Segfaults.20in.20CI.20are.20a.20pass.3F.html
---

<h2>Stream: <a href="https://leanprover-community.github.io/archive/stream/113488-general/index.html">general</a></h2>
<h3>Topic: <a href="https://leanprover-community.github.io/archive/stream/113488-general/topic/Segfaults.20in.20CI.20are.20a.20pass.3F.html">Segfaults in CI are a pass?</a></h3>

<hr>

<base href="https://leanprover.zulipchat.com/">

<head><link href="/style.css" rel="stylesheet"></head>

{% raw %}

<a name="274848375"></a>
<h4><a href="https://leanprover.zulipchat.com/#narrow/stream/113488-general/topic/Segfaults%20in%20CI%20are%20a%20pass%3F/near/274848375" class="zl"><img src="https://leanprover-community.github.io/archive/assets/img/zulip2.png" alt="view this post on Zulip"></a> Eric Wieser <a href="https://leanprover-community.github.io/archive/stream/113488-general/topic/Segfaults.20in.20CI.20are.20a.20pass.3F.html#274848375">(Mar 10 2022 at 15:14)</a>:</h4>
<p>I just made <a href="https://github.com/leanprover-community/mathlib/pull/12568">#12568</a>, which segfaults for me every time in gitpod. However, github reports that it built everything downstream of <a href="https://tqft.net/mathlib/algebra/order/monoid">file#algebra/order/monoid</a> in just 16s!</p>
<p>I assume what actually happened is that the build segfaulted in CI too, but CI considered that a pass?</p>



<a name="274849455"></a>
<h4><a href="https://leanprover.zulipchat.com/#narrow/stream/113488-general/topic/Segfaults%20in%20CI%20are%20a%20pass%3F/near/274849455" class="zl"><img src="https://leanprover-community.github.io/archive/assets/img/zulip2.png" alt="view this post on Zulip"></a> Eric Wieser <a href="https://leanprover-community.github.io/archive/stream/113488-general/topic/Segfaults.20in.20CI.20are.20a.20pass.3F.html#274849455">(Mar 10 2022 at 15:22)</a>:</h4>
<p>Ah, seems only the build step swallowed the error, lint and tests segfault too <a href="https://github.com/leanprover-community/mathlib/runs/5498018633?check_suite_focus=true">and report it</a></p>



<a name="274860314"></a>
<h4><a href="https://leanprover.zulipchat.com/#narrow/stream/113488-general/topic/Segfaults%20in%20CI%20are%20a%20pass%3F/near/274860314" class="zl"><img src="https://leanprover-community.github.io/archive/assets/img/zulip2.png" alt="view this post on Zulip"></a> Eric Wieser <a href="https://leanprover-community.github.io/archive/stream/113488-general/topic/Segfaults.20in.20CI.20are.20a.20pass.3F.html#274860314">(Mar 10 2022 at 16:33)</a>:</h4>
<p>Here's a <a href="https://leanprover-community.github.io/mwe.html">#mwe</a> for the segfault:</p>
<div class="codehilite" data-code-language="Lean"><pre><span></span><code><span class="kn">import</span> <span class="n">algebra.group.with_one</span>

<span class="c1">-- segfault without this line</span>
<span class="c1">-- attribute [semireducible] with_zero</span>

<span class="kd">def</span> <span class="n">segfault_please</span> <span class="o">:</span> <span class="n">with_zero</span> <span class="n">ℕ</span> <span class="bp">→</span> <span class="n">ℕ</span>
<span class="bp">|</span> <span class="mi">0</span> <span class="o">:=</span> <span class="mi">0</span>
<span class="bp">|</span> <span class="o">(</span><span class="n">n</span> <span class="o">:</span> <span class="n">ℕ</span><span class="o">)</span> <span class="o">:=</span> <span class="n">n</span>
</code></pre></div>



<a name="274870287"></a>
<h4><a href="https://leanprover.zulipchat.com/#narrow/stream/113488-general/topic/Segfaults%20in%20CI%20are%20a%20pass%3F/near/274870287" class="zl"><img src="https://leanprover-community.github.io/archive/assets/img/zulip2.png" alt="view this post on Zulip"></a> Gabriel Ebner <a href="https://leanprover-community.github.io/archive/stream/113488-general/topic/Segfaults.20in.20CI.20are.20a.20pass.3F.html#274870287">(Mar 10 2022 at 17:44)</a>:</h4>
<p><span aria-label="scream" class="emoji emoji-1f631" role="img" title="scream">:scream:</span></p>



<a name="274870477"></a>
<h4><a href="https://leanprover.zulipchat.com/#narrow/stream/113488-general/topic/Segfaults%20in%20CI%20are%20a%20pass%3F/near/274870477" class="zl"><img src="https://leanprover-community.github.io/archive/assets/img/zulip2.png" alt="view this post on Zulip"></a> Gabriel Ebner <a href="https://leanprover-community.github.io/archive/stream/113488-general/topic/Segfaults.20in.20CI.20are.20a.20pass.3F.html#274870477">(Mar 10 2022 at 17:46)</a>:</h4>
<blockquote>
<p>Ah, seems only the build step swallowed the error, lint and tests segfault too and report it</p>
</blockquote>
<p>Is this because we don't set pipefail in ci?</p>



<a name="274870866"></a>
<h4><a href="https://leanprover.zulipchat.com/#narrow/stream/113488-general/topic/Segfaults%20in%20CI%20are%20a%20pass%3F/near/274870866" class="zl"><img src="https://leanprover-community.github.io/archive/assets/img/zulip2.png" alt="view this post on Zulip"></a> Gabriel Ebner <a href="https://leanprover-community.github.io/archive/stream/113488-general/topic/Segfaults.20in.20CI.20are.20a.20pass.3F.html#274870866">(Mar 10 2022 at 17:49)</a>:</h4>
<p><a href="https://github.com/leanprover-community/lean/pull/695">lean#695</a></p>



{% endraw %}

<hr><p>Last updated: Jan 25 2023 at 00:06 UTC</p>